contracts/badger-sett/strategies/native/StrategyCurveLiquidityRewardGaugeBase.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.11;
+pragma experimental ABIEncoderV2;
+
+import "deps/@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "deps/@openzeppelin/contracts-upgradeable/math/SafeMathUpgradeable.sol";
+import "deps/@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
+import "deps/@openzeppelin/contracts-upgradeable/token/ERC20/SafeERC20Upgradeable.sol";
+
+import "interfaces/curve/ICurveFi.sol";
+import "interfaces/curve/ICurveGauge.sol";
+import "interfaces/curve/ICurveLiquidityRewardGauge.sol";
+import "interfaces/uniswap/IUniswapRouterV2.sol";
+
+import "interfaces/badger/IController.sol";
+import "interfaces/badger/IMintr.sol";
+import "interfaces/badger/IStrategy.sol";
+import "../BaseStrategy.sol";
+
+contract StrategyCurveGaugeBase is BaseStrategy {
```

Comment:
nit: think this needs to be renamed

Reply:
Intentionally committed as a separate strategy to prevent conflicts - I have no opinion on whether it should be used as the base strategy for all curve vaults or not.  It is intended to be adaptable to both (and yes the contract should be renamed.)

Reply:
I think my original strategies were confusing - the only difference between them ended up being the _add_liquidity function.
Really, we could just use parameters for this and have every strategy CRV gauge strategy be the same:
* Function variant to call (# of array elements)
* What position is _lpComponent we're using is at

Reply:
I'd propose moving it to a single strategy. I can implement in this new one and see what people think

Reply:
I'm on board consolidation here :).